### ISSUE TITLE
Should use go modules

### ISSUE THREAD

**Issue by swinslow** [spdx/tools-golang#12]
Currently, tools-golang doesn't have any dependencies, so dependency management (as well as how tools-golang is imported by downstream applications) hasn't really been thought through.

tools-golang should use go modules, see [[1]](https://github.com/golang/go/wiki/Modules), [[2]](https://blog.golang.org/modules2019). Specifically, it should be available for others to import as a go module.

I'm getting more familiar with the new go modules framework, but still new to it and would welcome input. It looks to me as though this should involve moving the various tools-golang packages out of the `v0/` directory, as I gather that go modules don't use `vX/` directories until at least a version 2 major release.

**Comment by RishabhBhatnagar** [comment]
tools-golang can still be used as a module.

```
package main


import (
    _"github.com/spdx/tools-golang/v0/builder"
)

func main(){
}
```

This is the way how we can use it as a module. Import has an underscore to allow compiling the code without using the imported library.

**Comment by rtgdk** [comment]
@RishabhBhatnagar That's useful. But I think what @swinslow wants is the directory layout of the tools-golang as a module. For eg., we need a go.mod file in the project to initialise the project as a module. Although right now, golang is going through a lot of changes on how the module dependency work. So we want to figure out a way to implement the project as a module which can be consistent with the next major release of the language.

**Comment by RishabhBhatnagar** [comment]
@rtgdk, Okay. 
So, according to [using-go-modules](https://blog.golang.org/using-go-modules), Starting in Go 1.13, module mode is the default for all development. 

But, the module feature is yet to be finalized. It will be finalized in Go 1.14 version. Initial variants of Go 1.14 are already released but they are unstable.

I don't think we should build on an unstable release. What should be done?
Should I create a go.mod file and make the project a module??

**Comment by swinslow** [comment]
Hi @RishabhBhatnagar, @rtgdk is right, that's what I'm hoping to do -- is to restructure spdx/tools-golang so that it follows the right practices for use as a Go module.

If you want to submit a PR that would add a go.mod file and make other appropriate changes, that would be great and I would gladly review it. I'm guessing that e.g. the v0/ directory should be removed, references to it in the other files and tests would need to be updated, etc.

It doesn't all need to happen at once but it would be great if we can move in that direction. Thank you for circling back and restarting discussion on this!

**Comment by rtgdk** [comment]
@RishabhBhatnagar Yeah, you can go ahead and try the `go mod init` command and change the directory layout as necessary. You would also need to take care of the different packages like 'builder' , 'idsearcher', etc. - they should be easily importable too. Also, checkout the changes done in `rdfreader` branch. 

Also, as @swinslow  said, you would need to update the test and some code for these changes to work. You don't have to do it all perfectly at once since even Go does not have a stable module dependency right now, but a working prototype would be a good way to start.

You expressed your interest in gsoc2020 on gitter, so this would be a good issue to get comfortable  with the codebase.

**Comment by RishabhBhatnagar** [comment]
Upon discussion in the gitter forum, I've got a clear idea that:
tools-golang is in v0 which is capable of dealing with spdx version 2.1.
Which commit should I mark as v0.0.0??

**Comment by rtgdk** [comment]
@RishabhBhatnagar  You can start by creating your own branch, make all the necessary changes and then mark that commit as v0.0.0

**Comment by swinslow** [comment]
Closed by #22, thanks again @RishabhBhatnagar!
